### PR TITLE
gh-106625 : Add missing code to tutorial 4.6 example

### PR DESCRIPTION
**Pull request by RustyNail**
This is a pull request for the [sample code in section 4.6](https://docs.python.org/ja/3/tutorial/controlflow.html#match-statements).

I ran the following based on the sample code and got an error.

The sample code in the tutorial may be an excerpt of some code.
However, given that we will try "Point(1, var)" immediately following, I believe the user will easily understand that there is a necessary dataclass import statement.

Executed code : 

```
class Point:
    x: int
    y: int

def where_is(point):
    match point:
        case Point(x=0, y=0):
            print("Origin")
        case Point(x=0, y=y):
            print(f"Y={y}")
        case Point(x=x, y=0):
            print(f"X={x}")
        case Point():
            print("Somewhere else")
        case _:
            print("Not a point")

## Added code
p = Point(1, 1)
```

Errors that occurred :

```
    p = Point(1, 1)
        ^^^^^^^^^^^
TypeError: Point() takes no arguments
```

----
:books: Documentation preview :books:: https://cpython-previews--106623.org.readthedocs.build/

* Issue: gh-106625

### CODE REVIEW

File: Doc/tutorial/controlflow.rst [hunk 1]
```diff
@@ -306,6 +306,9 @@ If you are using classes to structure your data
 you can use the class name followed by an argument list resembling a
 constructor, but with the ability to capture attributes into variables::
 
+    from dataclasses import dataclass
+    @dataclass
+
```
Comment (ericvsmith) [inline Doc/tutorial/controlflow.rst]: The `@dataclass` line should be directly abouve the `class Point:` line.

**Review by ericvsmith — DISMISSED** [review]

File: Doc/tutorial/controlflow.rst (hunk 1 above)
Comment (terryjreedy) [inline Doc/tutorial/controlflow.rst]:
```suggestion
```

Comment (terryjreedy) [reply Doc/tutorial/controlflow.rst]: fixed

File: Doc/tutorial/controlflow.rst [hunk 2]
```diff
@@ -306,6 +306,8 @@ If you are using classes to structure your data
 you can use the class name followed by an argument list resembling a
 constructor, but with the ability to capture attributes into variables::
 
+    from dataclasses import dataclass
+    @dataclass
```
Comment (terryjreedy) [inline Doc/tutorial/controlflow.rst:310]:
```suggestion
    from dataclasses import dataclass
    
    @dataclass
```

**Comment by methane** [comment]
> However, given that we will try "Point(1, var)" immediately following, I believe the user will easily understand that there is a necessary dataclass import statement.

That example is after the statement "You can use positional parameters with some builtin classes that provide an ordering for their attributes (e.g. dataclasses)."

Using dataclass before the statemnt is bit misleading.

In your case, following code would work.

```py
p = Point()
p.x = 0
p.y = 42
where_is(p)
```

Of course, it should ugly. But the first example should demonstrate that named argument pattern like `Point(x=y, y=y)` will work for regular classes. So we shouldn't use dataclass here.

Instead, I think this makes the first example better.

```py
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y
```

With this Point class, user will understand that even typehint for attributes are not necessary for `Point(x=x, y=y)` pattern.

**Comment by terryjreedy** [comment]
Reading @methane 's comment after fixing the example as a dataclass, I agree.  The example follows "f you are using classes to structure your data you can use the class name followed by an argument list resembling a constructor, but with the ability to capture attributes into variables:".  To me also, this implies a regular class with a contructor (`__init__` method).  The mention of dataclasses follows the example and I read that as applying to further examples.

**Comment by terryjreedy** [comment]
@RustyNail I already made the change.  Note that dunder names with double underscores need to be surrounded by backticks to display properly.  Otherwise the underscores are seen as markup.  Compare 'init' with bolder '__init__' with `__init__`.

**Comment by RustyNail** [comment]
@terryjreedy 
Thanks for the update.
I too understood your guys intent to use constructor initialization better.

**Comment by ericvsmith** [comment]
> Reading @methane 's comment after fixing the example as a dataclass, I agree.

I agree, too.

**Review by ericvsmith — APPROVED** [review]

**Pull request merged**